midas/algorithms/bayesian_optimization.py

```diff
@@ -0,0 +1,73 @@
+from skopt import Optimizer as skOptimizer
+from skopt.space import Real, Integer, Categorical
+
+class Bayesian_Optimization():
+   #!TODO: Finish docstrings (put in parameters)
+    """
+    Class for performing optimization using Bayesian Optimization, utilizing scikit-optimize python library
+
+    Written by Cole Howard, 10/21/2024
+    """
+    def __init__(self, input):
+        self.input = input
+        self.optimizer = skOptimizer(
+            dimensions = self.parse_dimensions(),
+            n_initial_points = 0,
+            acq_func = self.input.acquisition_function
+        )
+
+    def parse_dimensions(self):
```

Comment:
Understood. I plan to go back through my code at some point and thoroughly comment everything. The reason I have a parse_dimensions function is because the bayesian optimization needs to know what values each individual and chromosome can take. This library specifically needs it in a specific format, where each chromosome has a list of what values it can take, preceded by the value type (integer, real, or categorical). This function reads the genome and formats the possible values in this way for the optimizer object to understand. I hope this helps.